presto-main/src/main/java/com/facebook/presto/operator/WindowOperator.java

```diff
@@ -58,23 +65,37 @@ public WindowOperatorFactory(
                 List<Integer> outputChannels,
                 List<WindowFunctionDefinition> windowFunctionDefinitions,
                 List<Integer> partitionChannels,
+                List<Integer> preGroupedChannels,
                 List<Integer> sortChannels,
                 List<SortOrder> sortOrder,
+                int preSortedChannelPrefix,
```

Comment:
I did it this way to make it more clear as to what is valid input here. It doesn't make sense to pass columns in that are not part of a contiguous prefix.